tlatools/org.lamport.tlatools/src/tlc2/debug/TLCSourceBreakpoint.java

```diff
@@ -55,8 +68,95 @@ public TLCSourceBreakpoint(final String module, final SourceBreakpoint s) {
 			}
 		}
 		hits = h;
+		this.Condition = generateCondition(semanticRoot, s.getCondition());
 	}
-	
+
+	/**
+	 * Given a spec and an unparsed expression, build an operator that can be
+	 * evaluated within the context of that spec.
+	 * @param semanticRoot The root of the spec's semantic parse tree.
+	 * @param conditionExpr The unparsed expression.
+	 * @return A fully-initialized operator, or null if parsing failed.
+	 */
+	private static OpDefNode generateCondition(final ModuleNode semanticRoot, final String conditionExpr) {
+		if (null == conditionExpr || conditionExpr.isBlank()) {
+			return null;
+		}
+
+		// Find a placeholder identifier that has not yet been defined in
+		// the module.
+		final String candidatePattern = "__BreakpointExpr__%s";
+		String bpOpName = null;
+		long suffix = 0L;
+		do {
+			bpOpName = String.format(candidatePattern, Long.toString(suffix));
+			suffix++;
+		} while (semanticRoot.getOpDef(bpOpName) != null);
+
+		// Then, send expression through syntax parser; wrap expression
+		// in module to make syntax parser happy, then throw away the
+		// encapsulating module. Future syntax parser updates might allow
+		// parsing the operator definition directly.
+		final String wrapper = "---- MODULE Ignore ----\n%s == %s\n====";
```

Comment:
Perhaps, e.g. `Ignore21384ylshfutfaoug2` because `Ignore` could also be a user spec.

Reply:
The Ignore name will never make it to semantic checker so it won't be checked for conflicts; the syntax parser doesn't check for name collisions.

Reply:
Please add a corresponding note to the comment.